front/ui-components/src/stateful/Balance.tsx

```diff
@@ -26,12 +26,12 @@ export function Balance(props: BalanceProps): React.ReactElement {
     orientation = 'horizontal',
     ...rest
   } = props;
-  const { api, isReady } = useContext(ApiContext);
+  const { api, isApiReady } = useContext(ApiContext);
   const [allBalances, setAllBalances] = useState();
   const [allStaking, setAllStaking] = useState();
 
   useEffect(() => {
-    if (isReady) {
+    if (isApiReady) {
```

Comment:
I find it weird that we have to do this everywhere, since none of these should load in the first place before the `isApiReady` is set to true in the ContextGate.

Reply:
> since none of these should load in the first place before the isApiReady is set to true in the ContextGate

Hmm, if we put this in ui-components and expose this package to npm, then I don't think we should assume that the end-user will user this component inside a ContextGate when isApiReady is set to true. Just to be sure, I'd leave this check here.

In light-ui, if we use internal components that are behind a Gate, then yeah we can remove it.

Reply:
Edit: but also, as we talked before, a better way would simply only expose pure/stateless components in ui-components